velox/exec/DistinctAggregations.cpp

```diff
@@ -185,6 +199,47 @@ class TypedDistinctAggregations : public DistinctAggregations {
     return input->template asUnchecked<RowVector>()->children();
   }
 
+  void extractForSpill(folly::Range<char**> groups, VectorPtr& result) const {
+    auto* arrayVector = result->as<ArrayVector>();
+    arrayVector->resize(groups.size());
+
+    auto* rawOffsets =
+        arrayVector->mutableOffsets(groups.size())->asMutable<vector_size_t>();
+    auto* rawSizes =
+        arrayVector->mutableSizes(groups.size())->asMutable<vector_size_t>();
+
+    size_t totalBytes = 0;
+    vector_size_t offset = 0;
+    for (auto i = 0; i < groups.size(); ++i) {
+      auto* accumulator =
+          reinterpret_cast<AccumulatorType*>(groups[i] + offset_);
+      rawSizes[i] = accumulator->maxSpillSize();
```

Comment:
This code looks a bit strange. I assume SetAccumulator to contains a list of rows (distinct inputs for the aggregate function). Hence, rawSizes[i] would be the number of distinct rows stored in the accumulator. Hence, why the method that return this is called "maxSpillSize()" and not "size()"? And why is the same method used to update totalBytes? Would you clarify?

Reply:
maxSpillSize because when we are serializing each group the size is an upper bound in maxSpillSize and we serialize we have the actual count. The difference is that we are using an std::optional on the nullBytesIndex and if it is not set, we just serialize a char-size telling it is not set. If it is set, we serialize the char-size with 1 and the actual value. Pedantic, but was trying to be precise. Alternatively, we can change it to always serialize this std::optional regardless if it is set (say, with a 0 for the actual value) so the sizes are always exact. Would that be preferred here?